cmd/commands/client/command_client.go

```diff
@@ -42,8 +43,9 @@ func NewCommandWith(
 
 	identityManager := identity.NewIdentityManager(keystoreInstance)
 
-	dialogEstablisherFactory := func(myID identity.Identity) communication.DialogEstablisher {
-		return nats_dialog.NewDialogEstablisher(myID, identity.NewSigner(keystoreInstance, myID))
+	dialogFactory := func(consumerID, providerID identity.Identity, contact dto.Contact) (communication.Dialog, error) {
+		dialogEstablisher := nats_dialog.NewDialogEstablisher(consumerID, identity.NewSigner(keystoreInstance, consumerID))
```

Comment:
Why new `dialogEstablisher` struct is created for each dialog?

Reply:
That's because when connecting we can provide different identities